File: FrameworkDemo/app/src/main/java/me/longluo/fwk/MainActivity.java
package me.longluo.fwk;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import me.longluo.fwk.bluetooth.BluetoothActivity;
import me.longluo.fwk.wifi.WifiActivity;


public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private Button mBtnBluetooth;

    private Button mBtnWifi;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();
    }

    private void initView() {
        mBtnBluetooth = findViewById(R.id.btn_bluetooth);
        mBtnBluetooth.setOnClickListener(this);

        mBtnWifi = findViewById(R.id.btn_wifi);
        mBtnWifi.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        int viewId = view.getId();

        if (viewId == R.id.btn_wifi) {
            startActivity(new Intent(this, WifiActivity.class));
        } else if (viewId == R.id.btn_bluetooth) {
            startActivity(new Intent(this, BluetoothActivity.class));
        }
    }
}

File: socket/TcpServer/app/src/main/java/me/longluo/tcpserver/MainActivity.kt
package me.longluo.tcpserver

import android.content.Intent
import android.os.Build
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_empty)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            startForegroundService(Intent(applicationContext, TcpServerService::class.java))
        } else {
            startService(Intent(applicationContext, TcpServerService::class.java))
        }
    }
}